exercises/lesson_05/breach/lab

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+. data/tests/util.sh
+
+BASENAME=$(basename $0)
+TEMPFILE="$(mktemp -p /tmp ${BASENAME}XXX)"
+G1="\e[1;32m"
+G2="\e[0m"
+R1="\e[1;91m"
+R2="\e[0m"
+
+showHelp()
+{
+    echo "usage: lab [-h|--help] setup|intro|grade|clean"
+}
+
+
+checkEUID()
+{
+    if [[ $(id -ru) -ne 0 ]]; then
+        echo "$BASENAME: $1 must be run as root"
+        exit 1
+    fi
+}
+
+
+setup()
+{
+    for package in epel-release ansible; do
+        rpm -qa | grep $package &>/dev/null
+        if [[ $? -ne 0 ]]; then
+            yum install -y $package
+            echo $package >> $TEMPFILE
+        fi
+    done
+
+    cd data/ansible
+    ansible-playbook setup.yml
+    touch bootstrapped
+}
+
+
+ok()
+{
+    echo -e "${G1}PASS${G2}"
+}
+
+
+fail()
+{
+    if [[ -z ${1+x} ]]; then
+        RES=1
+    fi
+    echo -e "${R1}FAIL${R2} $1"
+}
+
+
+grade() {
+
+    if [[ ! -e ./data/ansible/bootstrapped ]]; then
+        echo "$BASENAME: 'setup' must be run first"
+        exit 1
+    fi
+
+    RET=0
+    for test_ in data/tests/[0-9]*; do
+        "$test_" || RES=1
+    done
+    return $RES
+}
+
+
+cleanup()
+{
+    cd data/ansible
+    ansible-playbook cleanup.yml
+
+    while read line; do
+        yum remove -y $line
+    done < $TEMPFILE
```

Comment:
If I understand correctly, student is able to delete this file. Is it save to store data so transparently in /tmp? If student deletes TEMPFILE, script won't clear the system properly.

Reply:
True, I'll put it in /var